adapter/internal/oasparser/envoyconf/routes_with_clusters.go

```diff
@@ -1391,5 +1391,5 @@ func getDefaultResourceMethods(apiType string) []string {
 
 func getDefaultVersionBasepath(basePath string, version string) string {
 	context := strings.ReplaceAll(basePath, "/"+version, "")
```

Comment:
Shall we fix this logic such that only the trailing part will be swapped?
I think we could use LastIndexOf and then pick a substring.

Reply:
This is the `getDefaultVersionBasepath`. This is correct right?

Reply:
Let's assume the basePath is `/foo/v2` and version is `/v2`. So the complete path would be `/foo/v2/v2` in that case, v2 would be replaced twice right?
This is an edge case actually, but I have seen people doing that still.

Reply:
+1

Reply:
fixed with https://github.com/wso2/product-microgateway/pull/2878/commits/3fb46d23794f856dd8102d2b17b8172ab0f13a37